sympy/concrete/tests/test_sums_products.py

```diff
@@ -265,7 +265,7 @@ def test_geometric_sums():
     assert result.is_Float
 
     result = Sum(0.25**n, (n, 1, oo)).doit()
-    assert result == S(1)/3
+    assert abs(result - 1.0/3) < 1e-11
```

Comment:
Why not `assert result == 1/3.`?

Reply:
I don't know what the precision is although I think something like `assert result == Float(1/3.0, precision=result._prec)` would work.

Reply:
BTW, I did try `result == 1/3.0` first but that failed. I assume this is because the precision of the result was not exactly 53 (but didn't check).

Reply:
Also, `1/3` should be the same as `1/3.0` in the test file, shouldn't it? It only failed for Py 2.7 as you can see in [this run](https://travis-ci.org/sympy/sympy/jobs/533019219).

Reply:
(Or maybe that was my mistake to use 1/3 instead of `1/3.` if truedivision was not imported.

Reply:
I assume that the result is calculated as `Float(0.25)/(1-Float(0.25))` which happens to work out the same as `1/3.` or `Float(1/3.)`.

Reply:
It's a truedivision situation: the test was formerly S(1)/3; I changed it to 1/3 (which passed locally because I was using Py 3) but failed under Travis Py 2.7. I changed it to `1/3.0` or similar and it works.

Reply:
The file ought to have `from __future__ import division` to avoid issues like this. With Python 2 support only hanging around for a few more months it might not be worth bothering with this, but we really should have made the standard `__future__` imports (`division` and `print_function`) required in the code quality checker.

Reply:
I think it's best not to have from `__future__` import division. Actually at some point I removed that from all test_ files. This example is different but mostly it just hides sloppy test code that uses floats when it should be using rational.

Reply:
If that is the goal, we need a more robust way of testing for that, because it will be moot once we drop Python 2 support. Not to mention it won't guard against cases like `4/2`, and possibly other subtleties where the test wouldn't fail in Python 2 even with integer division. Should we have a code quality guard against int/int, requiring an explicit float/int if that is what is intended (like `1./2`)? 

The `__future__.division` import generally helps prevent bugs from only testing things in Python 3. As I said, it's not super important since we are dropping Python 2 soon anyway, and most code should be tested for real in the test suite, which ought to catch any such errors as well.